### PR TITLE
docs(language-service): Fix quick info URLs

### PR DESCRIPTION
**Pull request by ViktorTyultinov**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #56410

## What is the new behavior?
Proper redirection to the new angular website from the quick info 

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/language-service/src/quick_info_built_ins.ts [hunk 1]
```diff
@@ -137,80 +137,80 @@ const BUILT_IN_NAMES_TO_DOC_MAP: {
 } = {
   '@defer': {
     docString: `A type of block that can be used to defer load the JavaScript for components, directives and pipes used inside a component template.`,
-    links: ['[AIO Reference](https://next.angular.io/api/core/defer)'],
+    links: ['[AIO Reference](https://angular.dev/guide/defer#defer)'],
```
Comment (atscott) [inline packages/language-service/src/quick_info_built_ins.ts:140]:
```suggestion
    links: ['[Reference](https://angular.dev/guide/defer#defer)'],
```

nit: Remove AIO as well from the text

**Review by atscott — CHANGES_REQUESTED** [review]

File: packages/language-service/src/quick_info_built_ins.ts (hunk 1 above)
Comment (ViktorTyultinov) [reply packages/language-service/src/quick_info_built_ins.ts]: I just did. I also made the links one bit more specific

**Review by atscott — APPROVED** [review]

**Comment by JeanMeche** [comment]
Can you fix the formatting by running `yarn ng-dev format files packages/language-service/src/quick_info_built_ins.ts`

**Comment by ViktorTyultinov** [comment]
@JeanMeche It should be fine now. Thank you for the guidance!

**Review by atscott — APPROVED** [review]

**Review by AndrewKushnir — APPROVED** [review]

**Comment by AndrewKushnir** [comment]
This PR was merged into the repository by commit c698668a522e927d3a90fdf3494747930911d106.

The changes were merged into the following branches: main, 18.0.x

**Pull request closed without merging**